src/channels/breakout/index.tsx

```diff
@@ -0,0 +1,303 @@
+		if (!app.current || !balls.current || !paddle.current || !crtFilter.current) return;
+
+		crtFilter.current.time += 0.1;
+		crtFilter.current.seed = Math.random();
+
+		if (breakout.balls.length > balls.current.length) {
+			balls.current.push(
+				...Array.from({ length: breakout.balls.length - balls.current.length }, () => {
+					const graphics = new PIXI.Graphics(BALL_TEMPLATE.geometry);
+					app.current!.stage.addChild(graphics);
+					return graphics;
+				}),
+			);
+		}
+
+		balls.current
+			.splice(breakout.balls.length, balls.current.length - breakout.balls.length)
+			.forEach((ballToRemove) => {
+				ballToRemove.destroy();
+				app.current!.stage.removeChild(ballToRemove);
+			});
+
+		if (breakout.blocks.length > blocks.current.length) {
+			blocks.current.push(
+				...Array.from({ length: breakout.blocks.length - blocks.current.length }, (_, i) => {
+					const graphics = new PIXI.Graphics(
+						BLOCK_TEMPLATES.get(breakout.blocks[blocks.current.length + i]?.color ?? Color.RED)?.geometry,
+					);
+					app.current!.stage.addChild(graphics);
+					return graphics;
+				}),
+			);
+		}
+
+		breakout.blocks.forEach((block, i) => {
+			blocks.current[i].setTransform(block.x, block.y);
+			blocks.current[i].visible = !block.destroyed;
+		});
+
+		breakout.balls.forEach((ball, i) => {
+			balls.current[i].setTransform(ball.x, ball.y);
+		});
+
+		paddle.current.setTransform(breakout.paddle.x, breakout.paddle.y);
+
+		if (breakout.isLevelClear()) {
+			// Clear all the balls.
+			breakout.balls.length = 0;
+
+			// Reset all the blocks;
+			breakout.blocks.forEach((block) => (block.destroyed = false));
+
+			// We won. Give us a starter ball.
+			emitBall();
+
+			resetting.current = false;
+		}
+
+		return () => {
+			crtFilter.current?.destroy();
+			crtFilter.current = null;
+
+			balls.current.forEach((ball) => {
+				if (!ball.destroyed) ball.destroy();
+				app.current!.stage.removeChild(ball);
+			});
+			balls.current.length = 0;
+
+			blocks.current.forEach((block) => {
+				if (!block.destroyed) block.destroy();
+				app.current!.stage.removeChild(block);
+			});
+			blocks.current.length = 0;
+
+			if (paddle.current) {
+				paddle.current.destroy();
+				app.current!.stage.removeChild(paddle.current);
+			}
+
+			if (backdrop.current) {
+				backdrop.current.destroy();
+				app.current!.stage.removeChild(backdrop.current);
+			}
+		};
+	});
+
+	// Create initial filters/backdrop
+	useEffect(() => {
+		if (!app.current || crtFilter.current) return;
+		crtFilter.current = new CRTFilter({
+			vignetting: 0.2,
+			time: 0,
+			lineWidth: 1,
+			lineContrast: 0.3,
+			noise: 0.1,
+		});
+
+		app.current.stage.filters = [crtFilter.current];
+
+		backdrop.current = new PIXI.Graphics();
+
+		backdrop.current.beginFill(Color.LIGHT_GRAY);
+		backdrop.current.drawRect(breakout.bounds.x, breakout.bounds.y, breakout.bounds.width, breakout.bounds.height);
+
+		app.current.stage.addChild(backdrop.current);
+	});
```

Comment:
No dependency array. This will execute everytime this component re-renders.

Reply:
Thanks. I'm a total React n00b, so I definitely need some help to spot stuff like this.